google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITBaseTest.java

```diff
@@ -48,6 +65,45 @@ public void before() {
 
     firestoreOptions = optionsBuilder.build();
     firestore = firestoreOptions.getService();
+    primeBackend();
+  }
+
+  public void primeBackend() throws Exception {
+    if (!backendPrimed) {
+      backendPrimed = true;
+      CompletableFuture<Void> watchInitialized = new CompletableFuture<>();
+      CompletableFuture<Void> watchUpdateReceived = new CompletableFuture<>();
+      DocumentReference docRef = firestore.collection(autoId()).document();
+      ListenerRegistration listenerRegistration =
+          docRef.addSnapshotListener(
+              (snapshot, error) -> {
+                assertNull(error);
```

Comment:
The assert is swallowed. If error is ever not null, nothing is propagated.

I don't know how best to handle this. There are better testing frameworks than vanila JUnit4 for asynchronous testing. I like when the test method receives a future `primeBackend(future)` that can be called asynchronously to indicate success or failure. That would make this a lot easier.

A rubber bandaid solution would be to do something like the following:

```
if (error != null) {
  logger.log(Level.SEVERE, "Prime backend received error in snapshot listener.", error);
  if (!watchInitialized.isDone()) {
    watchInitialized.completeExceptionally(error);
  } else if (!watchUpdateReceived.isDone()) {
    watchUpdateReceived.completeExceptionally(error);
  }
}
```